From ede893a90870c31d8b374978d524fddf6f3aac81 Mon Sep 17 00:00:00 2001
From: Asad Mehmood <asad78611@googlemail.com>
Date: Thu, 17 Oct 2013 16:21:42 +0100
Subject: [PATCH] GnomeRR: gnome_rr_output_get_min_backlight_step

Allows gnome-settings-daemon to read the min_backlight_step
so that it can support displays with less than 10 steps
---
 libgnome-desktop/gnome-rr.c | 15 +++++++++++++++
 libgnome-desktop/gnome-rr.h |  1 +
 2 files changed, 16 insertions(+)

diff --git a/libgnome-desktop/gnome-rr.c b/libgnome-desktop/gnome-rr.c
index 5e011d3..88df060 100644
--- a/libgnome-desktop/gnome-rr.c
+++ b/libgnome-desktop/gnome-rr.c
@@ -77,6 +77,7 @@ struct GnomeRROutput
     char *              edid_file;
 
     int                 backlight;
+    int                 min_backlight_step;
 
     gboolean            is_primary;
     gboolean            is_presentation;
@@ -1271,6 +1272,7 @@ output_initialize (GnomeRROutput *output, GVariant *info)
     g_variant_lookup (properties, "height-mm", "i", &output->height_mm);
     g_variant_lookup (properties, "display-name", "s", &output->display_name);
     g_variant_lookup (properties, "backlight", "i", &output->backlight);
+    g_variant_lookup (properties, "min-backlight-step", "i", &output->min_backlight_step);
     g_variant_lookup (properties, "primary", "b", &output->is_primary);
     g_variant_lookup (properties, "presentation", "b", &output->is_presentation);
 
@@ -1446,6 +1448,19 @@ gnome_rr_output_get_backlight (GnomeRROutput *output)
 }
 
 /**
+ * gnome_rr_output_get_min_backlight_step:
+ *
+ * Returns: The minimum backlight step available in percent
+ */
+int
+gnome_rr_output_get_min_backlight_step (GnomeRROutput *output)
+{
+    g_return_val_if_fail (output != NULL, -1);
+    
+    return output->min_backlight_step;
+}
+
+/**
  * gnome_rr_output_set_backlight:
  * @value: the absolute value which is 0 >= this <= 100
  *
diff --git a/libgnome-desktop/gnome-rr.h b/libgnome-desktop/gnome-rr.h
index a0d42f7..0443342 100644
--- a/libgnome-desktop/gnome-rr.h
+++ b/libgnome-desktop/gnome-rr.h
@@ -154,6 +154,7 @@ void            gnome_rr_output_get_physical_size  (GnomeRROutput         *outpu
                                                     int                   *height_mm);
 
 gint            gnome_rr_output_get_backlight      (GnomeRROutput         *output);
+gint            gnome_rr_output_get_min_backlight_step(GnomeRROutput      *output);
 gboolean        gnome_rr_output_set_backlight      (GnomeRROutput         *output,
                                                     gint                   value,
                                                     GError                **error);
-- 
1.8.4.1

